llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -3371,7 +3392,7 @@ bool SelectionDAGLegalize::ExpandNode(SDNode *Node) {
     Results.push_back(ExpandInsertToVectorThroughStack(SDValue(Node, 0)));
     break;
   case ISD::CONCAT_VECTORS:
-    Results.push_back(ExpandVectorBuildThroughStack(Node));
+    Results.push_back(ExpandConcatVectors(Node));
```

Comment:
My guess is that we'd want to do something like:
```suggestion
    if (TLI.isLegalOrCustom(ISD::EXTRACT_VECTOR_ELT, ElementValueType)
        Results.push_back(ExpandConcatVectors(Node));
    else
        Results.push_back(ExpandVectorBuildThroughStack(Node));
```

Reply:
Unfortunately it doesn't seem like `isOperationLegalOrCustom` works. I might be wrong, but in `Target/AArch64/AArch64TargetLowering.cpp`, it looks like `EXTRACT_VECTOR_ELT` is either legal or custom. Would it be possible to make the necessary change there?

Making the check `TLI.isOperationLegal` also doesn't change the test case arsenm brought up.